store.go

```diff
@@ -21,7 +21,7 @@ import (
 // See CookieStore and FilesystemStore for examples.
 type Store interface {
 	// Get should return a cached session.
-	Get(r *http.Request, name string) (*Session, error)
+	Get(r *http.Request, name string) (*Session, *http.Request, error)
```

Comment:
Ditto. Breaking the `Store` interface is a major break - it requires third-party store to update, which may be out of the control of the package user.